src/api/app/assets/javascripts/webui/canned_responses.js

```diff
@@ -1,10 +1,15 @@
 function setupCannedResponses() { // jshint ignore:line
   $('[data-canned-controller]').on('click', '[data-canned-response]', function(e) {
     $(e.target).closest('[data-canned-controller]').find('textarea').val(e.target.dataset.cannedResponse);
-    // TODO: adapt the following line when Decision#kind is replaced with Decision#type
-    // Set `cleared` by default for canned responses with decision_kind nil
-    let kind = e.target.dataset.decisionKind === 'favored' ? 'favor' : 'cleared';
-    $(e.target).closest('[data-canned-controller]').find('#decision_kind').val(kind);
+    // let's make sure this exists first, to avoid errors
+    if ($(e.target).closest('[data-canned-controller]').find('#decision_type').length !== 0) {
```

Comment:
Why this would not exist?

Reply:
Because in the dropdown we are displaying canned responses with decision_type = nil. Those responses which are meant to be used in normal comments, not only in decision-related ones (so they aren't cleared or favored).

Anyway, that line is already merged, as it's part of the main PR. This PR is based on that but I haven't rebased on master yet. I'm waiting for other PRs to be merged too.